assets/activation-scripts/activate.d/zsh

```diff
@@ -72,8 +72,8 @@ eval "$($_gnused/bin/sed -e 's/^/export /' -e 's/$/;/' $_add_env)"
 # HISTFILE and SHELL_SESSION_DIR variables, so define them up-front here
 # so that our env variable replay feature can fix the damage caused by
 # their /etc/zshrc* files.
-export HISTFILE=$HOME/.zsh_history
-export SHELL_SESSION_DIR=$HOME/.zsh_sessions
+export HISTFILE="${HISTFILE:-${ZDOTDIR:-$HOME}/.zsh_history}"
```

Comment:
Won't ZDOTDIR be set to something in the nix store? We don't want to fall back to that

Reply:
(feel free to dismiss my review if when we get to this file ZDOTDIR is set to something mutable, I've also lost track of the order of initialization)

Reply:
we reset ZDOTDIR to the original user one (or unset it if it wasnt set to begin with) earlier in the script:
https://github.com/flox/flox/blob/18ba811de3cdec02e6f8b02d8a84113c1c9778e3/assets/activation-scripts/activate.d/zsh#L20-L28